src/main/kotlin/org/rust/lang/core/psi/impl/mixin/RsArrayExprMixin.kt

```diff
@@ -0,0 +1,11 @@
+package org.rust.lang.core.psi.impl.mixin
+
+import com.intellij.lang.ASTNode
+import org.rust.lang.core.psi.RsArrayExpr
+import org.rust.lang.core.psi.RsExpr
+import org.rust.lang.core.psi.impl.RsCompositeElementImpl
+
+abstract class RsArrayExprMixin(node: ASTNode) : RsCompositeElementImpl(node), RsArrayExpr
```

Comment:
Why do we need mixin? Can we just define an extension method? 

We don't have a single place to define mixins for elements, but you could put this in org/rust/lang/core/psi/util/RustExprExtensions.kt.

I think it's wrong that we place extension methods in `impl/mixin` files. It's confusing to find an interface in an impl  package, and not every PSI element has a mixin. I'll try to organize this better.